README.md

```diff
@@ -1,16 +1,439 @@
+explicitly passed along with the function at the time of the function's call.
+
+The implicit way of passing a context with a function is something we have to
+memorize and accept as part of the nature of JavaScript.
+
+The tools for explicitly passing a context at function call-time are the
+methods `call`, `apply`, and `bind.`
+
+## Define `this`
+
+The JavaScript keyword `this` returns the current _execution context_. Whether
+that context was passed explicitly or implicitly, `this` returns it.
+
+We'll now identify places where the _execution context_ is implicitly set.
+
+### Access Implicitly-Set Global Object in a JavaScript Engine
+
+```javascript
+this //=> Window
+typeof this //=> "object"
+this === window //=> true
+```
+
+In a brand new Chrome console, try the code above. All the code you write in a
+JavaScript file can be thought of as "one big function" whose context is set to
+the `Window` object. Thus `this === window` is `true`.
+
+We can try the same thing out in NodeJS. If you have `node` installed on your
+computer, open the Node REPL by typing `node`.
+
+```js
+this === global //=> true
+```
+
+Node calls the default _execution context_ `global`, but otherwise it's a
+global object like `window`.
+
+> **STICKY PROBLEM**: Keen-eyed readers might be thinking: "But is `this ===
+> Window` also `true`?" It is not. The `window` property is part of the
+> `Window` global object; yes, the `Window` `Object` has a property that points
+> to itself.
+>
+> Due to the strictness of `===`, a property pointing to a thing is not the
+> same thing as the thing itself. As such `this === Window` is `false`, but `this
+> === window` is `true` _despite the fact_ that they both ultimately point to
+> the same thing.
+>
+> If all this strikes you as a bit fussy and uninteresting, it's not essential
+> understanding for the rest of this module. For more on this consult
+> [Window.window][w1].
+
+### Access Implicitly-Set Global Object in a Function Call
+
+***All function declarations and function expressions*** you have called to
+this point have also had an _execution context_, you just didn't know to ask
+for it with `this`.
+
+```js
+let contextReturner = function() {
+  return this
+}
+
+contextReturner() //=> window
+contextReturner() === window //=> true
+```
+
+When you invoked a function, the global context was _implicitly_ passed.
+
+JavaScript does this so that by default our functions can access any properties
+on that most-global object, `window` (or `global` in Node). Consider:
+
+```js
+let locationReturner = function() {
+  return this.location.host
+}
+
+locationReturner() //=> URL host serving this page e.g. developer.mozilla.org
+```
+
+Doesn't that seem _nice_ and handy to do? In an effort to be nice to us
+programmers, JavaScript's implementers made the global object available inside
+all functions.
+
+> **NEUROLINGUISTIC/ANTHROPOLOGICAL ASIDE**: Some people think that `this` is a
```

Comment:
I like this note, but it seems like it breaks the flow. Consider moving to the end of the section?